### PR TITLE
RD-13495 opentelemetry python contrib version

### PR DESCRIPTION
**Pull request by eugene-lumigo**
- **Version 1.26.0/0.47b0**

### CODE REVIEW

**Review by sagivoululumigo — CHANGES_REQUESTED** [review]
Remember to update the readme for supported runtimes.

Note: Don't manually edit the supported versions table, I think the automated version tester should take care of that

**Review by GuyMoses — CHANGES_REQUESTED** [review]
in the `README.md` file please state exactly when we deprecated python3.7 and state exactly the last known version to be supporting python3.7

File: .github/workflows/version-testing.yml [hunk 1]
```diff
@@ -42,11 +42,6 @@ jobs:
         with:
           python-version: 3.7
       - run: mkdir versions_artifacts
-      - uses: actions/download-artifact@v2
-        continue-on-error: true  # when there aren't changes the artifact won't be uploaded and this step will fail
-        with:
-          name: 3.7
-          path: versions_artifacts/3.7
```
Comment (sagivoululumigo) [inline .github/workflows/version-testing.yml:49]: Why remove this step?

File: src/lumigo_opentelemetry/__init__.py [hunk 2]
```diff
@@ -77,6 +77,16 @@ def auto_load(_: Any) -> None:
...
 def init() -> Dict[str, Any]:
+    """Initialize the Lumigo OpenTelemetry distribution."""
+    python_version = sys.version_info
+    # Check if the major version is 3 and the minor version is between 8 and 11
+    if python_version.major != 3 or not (8 <= python_version.minor <= 11):
+        logger.warning(
+            f"Unsupported Python version {python_version.major}.{python_version.minor}; "
+            "only Python 3.8 to 3.11 are supported."
+        )
+        return {}
```
Comment (sagivoululumigo) [inline src/lumigo_opentelemetry/__init__.py:88]: Put this inside a try catch block, so if in python 3.13 they change the interface of `sys.version_info` then the tracer won't crash.

If this block fails just log a warning saying that we failed to verify the python version.

File: src/lumigo_opentelemetry/instrumentations/fastapi/parsers/__init__.py [hunk 3]
```diff
@@ -51,7 +51,9 @@ async def new_otel_receive():  # type: ignore
         return new_otel_receive
 
     @staticmethod
-    def client_response_hook(span: Span, message: Dict[str, Any]) -> None:
+    def client_response_hook(
+        span: Span, scope: Dict[str, Any], message: Dict[str, Any]
+    ) -> None:
```
Comment (sagivoululumigo) [inline src/lumigo_opentelemetry/instrumentations/fastapi/parsers/__init__.py:56]: Why?

**Review by sagivoululumigo — COMMENTED** [review]
Overall good, left small comments

File: .github/workflows/version-testing.yml (hunk 1 above)
Comment (eugene-lumigo) [reply .github/workflows/version-testing.yml:49]: We discussed this

File: src/lumigo_opentelemetry/__init__.py (hunk 2 above)
Comment (sagivoululumigo) [reply src/lumigo_opentelemetry/__init__.py]: Don't only check for an `AttributeError` check for any error.
This is a fail safe for unexpected errors, so don't catch only one expected error...

File: .github/workflows/version-testing.yml (hunk 1 above)
Comment (sagivoululumigo) [reply .github/workflows/version-testing.yml:49]: OK

File: src/lumigo_opentelemetry/__init__.py (hunk 2 above)
Comment (sagivoululumigo) [reply src/lumigo_opentelemetry/__init__.py]: Approved the fix

File: src/lumigo_opentelemetry/instrumentations/fastapi/parsers/__init__.py (hunk 3 above)
Comment (sagivoululumigo) [reply src/lumigo_opentelemetry/instrumentations/fastapi/parsers/__init__.py:56]: Understood

**Review by sagivoululumigo — APPROVED** [review]
Approved

**Review by GuyMoses — APPROVED** [review]

**Pull request merged**